### PR TITLE
MULE-18871: Fix flaky test org.mule.test.module.extension.source.HeisenbergMessageSourceTestCase.sourceRestartedWithDynamicConfig

### PR DESCRIPTION
**Pull request by pabloperalta**
…enbergMessageSourceTestCase.sourceRestartedWithDynamicConfig

### CODE REVIEW

**Comment by pabloperalta** [comment]
--validate

**Comment by eze210** [comment]
--validate

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/source/HeisenbergMessageSourceTestCase.java [hunk 1]
```diff
@@ -331,6 +332,20 @@ protected void stopFlow(String flowName) throws Exception {
...
+  protected void requestFlowToStopAndWait(String flowName) throws Exception {
+    stopFlow(flowName);
+    checkFlowIsStopped(flowName);
+  }
+
+  protected void checkFlowIsStopped(String flowName) throws Exception {
+    flow = (Flow) getFlowConstruct(flowName);
+    check(FLOW_STOP_TIMEOUT, POLL_DELAY_MILLIS,
+          () -> {
+            assertThat("The flow did not stop in a reasonable amount of time", flow.getLifecycleState().isStopped(), is(true));
```
Comment (eze210) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/source/HeisenbergMessageSourceTestCase.java:344]: return flow.getLifecycleState().isStopped();

Comment (eze210) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/source/HeisenbergMessageSourceTestCase.java:344]: And move the error message to the probe description

**Review by eze210 — CHANGES_REQUESTED** [review]

**Review by eze210 — APPROVED** [review]

**Comment by pabloperalta** [comment]
--validate

**Comment by pabloperalta** [comment]
--validate

**Comment by pabloperalta** [comment]
--validate

**Comment by pabloperalta** [comment]
--validate

**Pull request merged**